packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-secret-manager/auth-secret-manager.ts

```diff
@@ -0,0 +1,84 @@
+import { $TSContext, spinner, stateManager } from 'amplify-cli-core';
+import aws from 'aws-sdk';
+import { getFullyQualifiedSecretName, oauthObjSecretKey } from './secret-name';
+
+/**
+ * Manages the state of OAuth secrets in AWS ParameterStore
+ */
+export class OAuthSecretsStateManager {
+  private static instance: OAuthSecretsStateManager;
+  private oAuthSecretKey: string;
+
+  static getInstance = async (context: $TSContext): Promise<OAuthSecretsStateManager> => {
+    if (!OAuthSecretsStateManager.instance) {
+      OAuthSecretsStateManager.instance = new OAuthSecretsStateManager(context, await getSSMClient(context));
+    }
+    return OAuthSecretsStateManager.instance;
+  };
+
+  private constructor(private readonly context: $TSContext, private readonly ssmClient: aws.SSM) {
```

Comment:
```suggestion
  private constructor(private readonly ssmClient: aws.SSM) {
```

Reply:
context is required in fetching ssm client from providers

Reply:
context is required to fetch ssm client

Reply:
right, but the client is passed to the constructor. The instance doesn't need a reference to context